### PR TITLE
rework dask cache

### PR DESCRIPTION
**Pull request by tlambert03**
# Description
This addresses #1205, by making sure that if a user has resized the dask cache manually, then it will not be changed.  And once a cache has been created, it can only be resized using `utils.resize_dask_cache`.  See the tests `test_dask_cache_resizing` and `test_prevent_dask_cache` for examples of what this is fixing.  Let's maybe leave #1205 open for input on tuning the default cache size (which this PR does not change)

Summary:
- `utils.resize_dask_cache(0)`, called _either_ before or after Layer creation will (permanently) disable the cache.
- cache can be resized up or down using `utils.resize_dask_cache(nbytes)` and it will stick.
- if someone actually deletes the `utils.dask_cache` attribute, we will lose the "memory" of the cache, and it will therefore get spun up again on the next Layer addition (but it won't break) ... so using the `resize_dask_cache` API is the way to go.

## Type of change

- [x] Bug-fix (non-breaking change which fixes an issue)

# How has this been tested?

- [x] added a couple new well-commented tests that describe/test the use case
- [x] all tests pass with my change

## Final checklist:
- [x] My PR is the minimum possible work for the desired functionality
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: napari/utils/misc.py [hunk 1]
```diff
@@ -292,8 +292,42 @@ def all_subclasses(cls: Type) -> set:
...
+        The fraction (from 0 to 1) of total memory to use for the dask cache.
+        by default, 50% of total memory is used.
+
+    Returns
+    -------
+    [type]
+        [description]
+    """
+    import psutil
+    from napari import utils
```
Comment (jni) [inline napari/utils/misc.py]: Relative import?

File: napari/utils/misc.py [hunk 2]
```diff
@@ -332,26 +366,31 @@ def resize_dask_cache(
     >>> cache.cache.total_bytes   # currently used bytes
     """
 
-    from napari.utils import dask_cache
+    from dask.cache import Cache
+    from napari import utils
```
Comment (jni) [inline napari/utils/misc.py]: Relative import?

File: napari/layers/_tests/test_dask_layers.py [hunk 3]
```diff
@@ -157,3 +161,66 @@ def dask_optimized_slicing(*args, **kwds):
...
+
+    v = viewer.ViewerModel()
+    dask_stack = delayed_dask_stack['stack']
+    # adding a new stack will not increase the cache size
+    v.add_image(dask_stack, multiscale=False, contrast_limits=(0, 1))
+    assert utils.dask_cache.cache.available_bytes == 0
+    # and the cache will not be populated
+    for i in range(3):
+        v.dims.set_point(0, i)
+    assert len(utils.dask_cache.cache.heap.heap) == 0
```
Comment (jni) [inline napari/layers/_tests/test_dask_layers.py:226]: 😂 this is quite a line but I’ll take it!

**Review by jni — APPROVED** [review]
Very nice @tlambert03 ! I think a couple of imports should be made relative, but otherwise this is good to go!

File: napari/layers/_tests/test_dask_layers.py (hunk 3 above)
Comment (tlambert03) [reply napari/layers/_tests/test_dask_layers.py:226]: yeah 😄 ... I don't think we'll ever need to access that directly.  but that's where the actual hashes are stored, for a rainy day!

**Comment by tlambert03** [comment]
cleaned up imports.  can be merged after tests pass

**Review by sofroniewn — APPROVED** [review]
LGTM!

**Pull request merged**